### PR TITLE
Adding ESLint to the project and solving the ESLint Errors

### LINKED ISSUE #293: Add Linting
Hello!
I need to implement linting feature to this repository. Can someone assign me this issue. So, that I can start working on it.

### PR DESCRIPTION
**Pull request by bhavik001**
Fixes #293 
### Description
This pull request addresses the issue related to setting up ESLint in the repository to improve and maintain a consistent code quality across the project. ESLint is a powerful static code analysis tool that helps identify and fix potential issues, enforcing a set of coding standards and best practices. Also, fixes some eslint error.

### Changes Made
- [x] Configure ESLint to the project
- [x] Package.json Updates.
- [x] Configured ESLint to run as part of the continuous integration (CI) process, ensuring that code quality is checked automatically with each pull request.

<sub>
By opening a pull request, I certify that I hold the intellectual property of the code I am submitting,
and I am granting the initial authors of WBO a perpetual, worldwide, non-exclusive, royalty-free, and irrevocable license to this code.
</sub>

### CODE REVIEW

**Comment by lovasoa** [comment]
Thanks for opening this. However, can you please switch this to a draft until the actual linting errors are fixed? Adding eslint doesn't make sense if we just litter the code with `eslint-disable-next-line` everywhere.

**Comment by bhavik001** [comment]
I am still working on it.

**Comment by lovasoa** [comment]
Yes, no problem !

**Comment by bhavik001** [comment]
Hey @lovasoa 
This is branch is ready for merge.

**Comment by lovasoa** [comment]
It looks like the pr introduces functional changes. We don't want to leave commented-out code.

**Comment by lovasoa** [comment]
You commented-out code. Either you made sure this code was not used, and you can remove it (and comment here about it) or it was actually used, and it should stay.

And we don't want to keep the eslint-disable-next-line...

**Comment by bhavik001** [comment]
Hey, I have remove all the lines which has eslint-disable-next-line and commented codes.
Please review the code and let me know if I have to make any changes in the code then.
This is the function I have removed form server/socket.js file:
```
// This is function is declared in board.js file and it is not used here
function generateUID(prefix, suffix) {
  var uid = Date.now().toString(36); //Create the uids in chronological order
  uid += Math.round(Math.random() * 36).toString(36); //Add a random character at the end
  if (prefix) uid = prefix + uid;
  if (suffix) uid = uid + suffix;
  return uid;
 }
```

File: server/check_output_directory.js [hunk 1]
```diff
@@ -14,7 +14,8 @@ async function get_error(directory) {
     return "does not exist";
   }
   if (!fs.statSync(directory).isDirectory()) {
-    error = "exists, but is not a directory";
+    // error = "exists, but is not a directory";
```
Comment (lovasoa) [inline server/check_output_directory.js:17]: let's avoid leaving commented-out code

File: server/check_output_directory.js [hunk 2]
```diff
@@ -28,16 +29,17 @@ async function get_error(directory) {
         "Check the permissions of the directory, and if needed change them so that " +
         `user with UID ${uid} has access to them. This can be achieved by running the command: chown ${uid}:${gid} on the directory`;
     } finally {
-      return err_msg;
+      console.log(err_msg);
```
Comment (lovasoa) [inline server/check_output_directory.js:32]: Wow! This changes the behavior ! 

Please avoid introducing changes like that before discussing them !

If you do feel some behavior change is required, let's discuss it together first, and add tests for it.

File: server/sockets.js [hunk 3]
```diff
@@ -29,17 +29,19 @@ function noFail(fn) {
...
   if (config.AUTH_SECRET_KEY) {
     // Middleware to check for valid jwt
     io.use(function (socket, next) {
       if (socket.handshake.query && socket.handshake.query.token) {
         jsonwebtoken.verify(
           socket.handshake.query.token,
           config.AUTH_SECRET_KEY,
-          function (err, decoded) {
-            if (err)
+          // function (err, decoded) {
```
Comment (lovasoa) [inline server/sockets.js:40]: let's avoid leaving commented-out code

File: server/sockets.js [hunk 4]
```diff
@@ -204,14 +212,6 @@ async function saveHistory(boardName, message) {
...
 }
 
-function generateUID(prefix, suffix) {
-  var uid = Date.now().toString(36); //Create the uids in chronological order
-  uid += Math.round(Math.random() * 36).toString(36); //Add a random character at the end
-  if (prefix) uid = prefix + uid;
-  if (suffix) uid = uid + suffix;
-  return uid;
-}
-
```
Comment (lovasoa) [inline server/sockets.js:214]: ?

File: tests/integration.js [hunk 5]
```diff
@@ -111,6 +111,7 @@ function testCursor(browser) {
       e.pageX = 150;
       e.pageY = 200;
       Tools.board.dispatchEvent(e);
+      done();
```
Comment (lovasoa) [inline tests/integration.js:114]: ?

**Review by lovasoa — CHANGES_REQUESTED** [review]

File: server/jwtauth.js [hunk 6]
```diff
@@ -24,8 +24,8 @@
  * @licend
  */
 
-(config = require("./configuration.js")),
-  (jsonwebtoken = require("jsonwebtoken"));
+const config = require("./configuration.js");
+// const jsonwebtoken = require("jsonwebtoken");
```
[the remark this answers is not in the document]
Comment (bhavik001) [reply server/jwtauth.js]: Check my latest commit

File: server/log.js [hunk 7]
```diff
@@ -9,8 +9,10 @@ const config = require("./configuration.js"),
 function parse_statsd_url(url) {
   const regex = /^(tcp|udp|statsd):\/\/(.*):(\d+)$/;
   const match = url.match(regex);
-  if (!match)
+  if (!match) {
     throw new Error("Invalid statsd connection string, doesn't match " + regex);
+  }
+  // eslint-disable-next-line no-unused-vars
```
[the remark this answers is not in the document]
Comment (bhavik001) [reply server/log.js]: check my latest commit

File: server/sockets.js [hunk 8]
```diff
@@ -204,13 +212,14 @@ async function saveHistory(boardName, message) {
...
-  return uid;
-}
+// This is function is declared in board.js file and it is not used here
+// function generateUID(prefix, suffix) {
+//   var uid = Date.now().toString(36); //Create the uids in chronological order
+//   uid += Math.round(Math.random() * 36).toString(36); //Add a random character at the end
+//   if (prefix) uid = prefix + uid;
+//   if (suffix) uid = uid + suffix;
+//   return uid;
+// }
```
[the remark this answers is not in the document]
Comment (bhavik001) [reply server/sockets.js]: Check my latest commit I have remove this function as this function is not used anywhere in the file and this function is created in the client-data/js/board.js file

File: server/check_output_directory.js (hunk 2 above)
Comment (bhavik001) [reply server/check_output_directory.js:32]: For above I am getting lint error: Unsafe usage of ReturnStatement.
Explaination:
 In JavaScript, using `return` in a `finally` block can lead to unexpected behavior, and it's generally considered unsafe. The `finally` block is designed for code that should be executed regardless of whether an exception is thrown or caught. Placing a `return` statement in a finally block can interfere with the intended flow.

File: server/sockets.js (hunk 4 above)
Comment (bhavik001) [reply server/sockets.js:214]: Check my latest commit I have remove this function as this function is not used anywhere in the file and this function is created in the client-data/js/board.js file

File: tests/integration.js (hunk 5 above)
Comment (bhavik001) [reply tests/integration.js:114]: In line 108 I was getting the below lint error:
'done' is declared but its value is never.
'done' is defined but never used. Allowed unused args must match /^_/u.
So, I have done some research and add done() in line 114.
Explaination:
In the context of an asynchronous function, the done function is often used as a callback to signal the completion of an asynchronous operation. It is a convention in testing frameworks, particularly in asynchronous testing, where you want to ensure that certain asynchronous tasks are completed before proceeding with the next step or completing the test.

You can check other functions in that file.

File: server/check_output_directory.js (hunk 2 above)
Comment (lovasoa) [reply server/check_output_directory.js:31]: Well, you cannot just remove the return statement. This changes the behavior of the function. In this case, the return was intended. You can refactor it, but not change the behavior. Even less when you are not adding tests.

File: tests/integration.js (hunk 5 above)
Comment (lovasoa) [reply tests/integration.js:114]: the function is synchronous, you don't need the done function at all, do you ?

Comment (bhavik001) [reply tests/integration.js:114]: I just have to add done() as you have add done in this line: `.execute(function (done) {`
```
function testCursor(browser) {
  return browser
    .execute(function (done) {
      Tools.setColor("#456123"); // Move the cursor over the board
      var e = new Event("mousemove");
      e.pageX = 150;
      e.pageY = 200;
      Tools.board.dispatchEvent(e);
      done();
    })
    .assert.cssProperty(
      "#cursor-me",
      "transform",
      "matrix(1, 0, 0, 1, 150, 200)",
    )
    .assert.attributeEquals("#cursor-me", "fill", "#456123");
}
```

if I remove done() from the line 114 then I have to remove it from line 108

**Pull request closed without merging**